test/test_torch.py

```diff
@@ -8068,6 +8068,23 @@ def generate_inputs(num_batches):
         finally:
             torch.set_num_threads(num_threads)
 
+# Test Python Array API Constants
```

Comment:
Instead of this add a test to TestTorchDeviceType like this:

```
# Validates that mathematical constants are defined properly, as required by 
#   the Python Array API (https://data-apis.org/array-api/latest/API_specification/constants.html)
@onlyCPU
def test_constants(self, device):
  self.assertEqual(torch.e, math.e, atol=0, rtol=0)
  self.assertEqual(torch.pi, math.pi, atol=0, rtol=0)
  self.assertEqual(torch.nan, math.nan, equal_nan=True)
  self.assertEqual(torch.inf, math.inf)
```

Reply:
No strong opinion, but I'm not sure how that makes it easier / better. Why do you think an additional test case is not a good solution here?

Reply:
Why add yet another test case when a test will do?

Reply:
Because `TestTorchDeviceType` has very little to do with the constants. There is nothing device generic that we need to check here.

Reply:
True, that's why we should decorate the test with `onlyCPU`. Ideally every test would use the test generator and be explicit about which devices it uses (or doesn't).

Reply:
As I said, I don't have a strong opinion here, so I'm backing down now. I just feel there is no sense in adding a test to and device generic test case that it feature wise doesn't belong too only to then only select a single device. 

> Ideally every test would use the test generator and be explicit about which devices it uses (or doesn't).

Do you mean in general throughout the whole test suite or just for this module?

Reply:
Throughout the entire test suite, because it's useful to understand what devices a test wants (or doesn't).